controllers/api.controller.js

```diff
@@ -40,3 +41,18 @@ exports.getArticles = (req, res, next) => {
     })
     .catch(next);
 };
+
+exports.getCommentsByArticleId = (req, res, next) => {
+  const { article_id } = req.params;
+  const promises = [fetchCommentsByArticleId(article_id)];
+
+  if (article_id) {
+    promises.push(checkArticleExist(article_id));
+  }
+
+  Promise.all(promises)
+    .then((comments) => {
```

Comment:
To make this cleaner you can destructure `comments` from the parameter. Array destructuring works in a similar way to object destructuring.

`[comments]`